WcaOnRails/app/views/competitions/_registration_requirements.html.erb

```diff
@@ -1,13 +1,23 @@
 <% if @competition.use_wca_registration.nil? %>
 <% elsif @competition.use_wca_registration %>
-  <%= t('competitions.competition_info.create_wca_account_html', here: link_to(t('common.here'), new_user_registration_path)) %>
-  <br>
-  <%= t('competitions.competition_info.claim_wca_id_html', here: link_to(t('common.here'), profile_claim_wca_id_path)) %>
-  <br>
-  <%= t('competitions.competition_info.register_html', here: link_to(t('common.here'), competition_register_path(@competition))) %>
-  <br>
+  <% if @current_user.nil? %>
+    <%= t('competitions.competition_info.create_wca_account_html', here: link_to(t('common.here'), new_user_registration_path)) %>
+    <br>
+    <%= t('competitions.competition_info.claim_wca_id_html', here: link_to(t('common.here'), profile_claim_wca_id_path)) %>
+    <br>
+  <% elsif !@current_user.wca_id && !@current_user.unconfirmed_wca_id %>
+    <%= t('competitions.competition_info.claim_wca_id_html', here: link_to(t('common.here'), profile_claim_wca_id_path)) %>
+    <br>
+  <% end %>
+  <% if current_page?(competition_register_path(@competition)) %>
```

Comment:
When we want to use a partial in different parts of the website and have it do slightly different things, we prefer passing arguments that change its behavior instead of having it check what page it's being used on. You can see an example of this with our time_limit_div partial:

- https://github.com/jfly/worldcubeassociation.org/blob/fix-documents-list/WcaOnRails/app/views/competitions/_time_limit_div.html.erb
- see usage here: https://github.com/jfly/worldcubeassociation.org/blob/9af3696f686a8baa6bbb6023e1ca3695a648c604/WcaOnRails/app/views/competitions/_events_tab.html.erb#L51